controllers/eventbus/installer/installer.go

```diff
@@ -54,16 +57,77 @@ func getLabels(bus *v1alpha1.EventBus) map[string]string {
 	}
 }
 
-// Uninstall function uninstalls the extra resources who were not cleaned up
-// when an eventbus was deleted. Most of the time this is not needed as all
+// Uninstall function will be run before the EventBus object is deleted,
+// usually it could be used to uninstall the extra resources who would not be cleaned
+// up when an EventBus is deleted. Most of the time this is not needed as all
 // the dependency resources should have been deleted by owner references cascade
 // deletion, but things like PVC created by StatefulSet need to be cleaned up
 // separately.
-func Uninstall(eventBus *v1alpha1.EventBus, client client.Client, natsStreamingImage, natsMetricsImage string, logger *zap.SugaredLogger) error {
+//
+// It could also be used to check if the EventBus object can be safely deleted.
+func Uninstall(ctx context.Context, eventBus *v1alpha1.EventBus, client client.Client, natsStreamingImage, natsMetricsImage string, logger *zap.SugaredLogger) error {
+	linkedEventSources, err := linkedEventSources(ctx, eventBus.Namespace, eventBus.Name, client)
```

Comment:
Check if there's any EventSource or Sensor connected, if true, return an error, which will re-queue the request.